### PR TITLE
move benchmark utils into torch namespace

### PR DESCRIPTION
**Pull request by robieta**
Move the timing utils to `torch.utils._benchmark`. I couldn't figure out how to get setuptools to pick it up and put it under `torch` unless it is in the `torch` directory. (And I think it has to be for `setup.py develop` anyway.)

I also modified the record function benchmark since `Timer` and `Compare` should always be available now.

### CODE REVIEW

**Review by ngimel — APPROVED** [review]
Cool! Can we add some smoketests to CI to make sure Timer, Compare and Fuzzer are importable and usable? What's not being tested breaks.

File: test/test_utils.py [hunk 1]
```diff
@@ -591,5 +592,42 @@ def test_import_hipify(self):
...
+        )
+
+        expected_results = [
+            (0.7821, 0.0536, 0.9888, 0.1949, 0.5242, 0.1987, 0.5094),
+            (0.7166, 0.5961, 0.8303, 0.005),
+        ]
+
+        for i, (tensors, _, _) in enumerate(fuzzer.take(2)):
+            x = tensors["x"]
+            self.assertTrue(
```
Comment (ngimel) [inline test/test_utils.py:628]: can you please use self.assertEqual here instead of torch.allclose in the statement?

Comment (robieta) [reply test/test_utils.py]: Done. I didn't realize assertEqual would take a non-scalar value. Thanks!

Comment (ngimel) [reply test/test_utils.py]: Not only non-scalar, it would recursively go over lists or tuples you send it.

**Comment by robieta** [comment]
I tweaked the type annotations since moving to `torch` means that MyPy now inspect the benchmark util. The remaining test failures are because of weird python import collisions, and I'm not sure if they're real.

**Comment by robieta** [comment]
The one build that is still broken is `pytorch_windows_vs2019_py36_cuda10.1_test2` because apparently numpy random can't accept seeds greater than int32 max on that platform. My inclination is just to disable the fuzzer test if `IS_WINDOWS` and `VC_YEAR`==2019, since this is a pretty marginal case. Thoughts?

**Comment by ngimel** [comment]
Yeah, disabling test sounds ok.

**Pull request closed without merging**